### DISCUSSION TITLE
TTS - Read letter by letter rather than word by word

### DISCUSSION THREAD

**Discussion by mrhunsaker** [monkeytypegame/monkeytype#4796] [Ideas]
I am working on modifying Monkeytype for use with my blind students. I was wondering if anyone could provide a bit of clarity to how best to approach this. I do want the program to be using words and not just spitting out random letters (which would be easy to implement by altering a language file to just have a list of letters).  Alternately I could put a flag to have the program read each letter after the previous had been typed in all modes, but that is even more elusive to me than the smaller question.

**Comment by Miodec** [comment]
The funbox change would be a bit more complicated because right now the funbox system only supports running scripts per word - no other funbox does something per letter. 

If you wanted to do it quickly, you could run `TTSEvent.dispatch(characterToSpeak)` probably somewhere inside the `handleChar` function or `updateWordElement`. You would need to also add it when the test is first started, and also when backspacing, moving to previous words and so on.

Let me know if it makes sense.
